rental.rb

```diff
@@ -0,0 +1,14 @@
+require './book'
+require './person'
+
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
+    book.rentals << self
+    person.rentals << self
```

Comment:
You have evidently put in a lot of work into this and so far you have done very well and most of the functions of the app are working. However, whenever I have to create a new rental, I get an error and it seems to be coming from line 12 here. The error states: ```undefined method `rentals' for #<Teacher:0x00007f529593c3a0 @id=2, @age="42", @name="teset2", @parent_permission=true, @specialization="math"> (NoMethodError)``` and the error seems to be triggered on line 133 of ```app.rb```. My thought is that there is no ```rentals``` method or variable in the Person class in person.rb and so trying to populate this person.rentals entity like you are trying to do here on line 12 is throwing an error. I am assuming that you are trying to populate a rentals array in the Person class so that you can display them when your user selects the option to list the rentals for a given person id and if so then I suggest you check for the implementation of the rentals array in the Person class. 

![image](https://github.com/Odongo006/OOP-School-Library/assets/107795045/93e1e3e2-2531-4127-9d62-ddb2a2b0a05e)